sympy/core/sympify.py

```diff
@@ -263,19 +263,16 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
 
     """
     try:
-        if a in sympy_classes:
+        if a.__sympy__:
             return a
-    except TypeError: # Type of a is unhashable
+    except AttributeError:  # a is not a sympy object
```

Comment:
Please don't catch AttributeError as it can hide bugs. It's better to do
```
    attr = getattr(a, '__sympy__', None)
    if attr is not None:
```
(like you do below for `__class__`)